src/views/Home.jsx

```diff
@@ -1,11 +1,24 @@
 import './Home.css';
+import { generateToken } from '@the-collab-lab/shopping-list-utils';
+import { addItem } from '../api';
+
+export function Home({ setListToken }) {
+	// upon button click generate new token and send it to parent component to store in state
+	// also send a hidden placeholder item to firestore to create the collection for this list token (will be filtered out so just setting next purchase to default value of 7)
+	const clickHandler = () => {
+		let newToken = generateToken();
+		setListToken(newToken);
+		addItem(newToken, {
+			itemName: 'placeholder',
+			daysUntilNextPurchase: 7,
+			hidden: true,
+		});
```

Comment:
I know firebase doesn't create the collection until there is a document, but once we setListToken() we are given back the name of the collection in local storage that will be used in firebase. 

Once we have this reference, we could just add an item to the list and it will populate the firebase database with the collection and item you have now inputed. 

here is the suggested change you could make. 

```suggestion
		collection(db, newToken);
```
once we add item it will update the database w

Reply:
Actually now that I look at it more I don't think that is necessary as well. 
Everything is handled in the addItem() component. 
```suggestion

```

Reply:
In this case the button should only create a token.